### PR TITLE
Add controller integration tests to the CI

### PR DESCRIPTION
**Pull request by JonCrowther**
## Summary

Added a new step to the CI to run the controller tests. I used `verify-generated-code-changes.yml` as a reference.

I also updated the `run_controller_tests.sh` script to remove an unused environment variable and to run in the right directory.

### CODE REVIEW

**Review by andreas-kupries — APPROVED** [review]

**Review by andreas-kupries — DISMISSED** [review]

File: .github/workflows/controller-test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+    - 'release/*'
+
+env:
+  MAIN_BRANCH: origin/master
```
Comment (raulcabello) [inline .github/workflows/controller-test.yml:11]: what's the `MAIN_BRANCH` env var doing? Do we really need it?

File: tests/controllers/run_controller_tests.sh [hunk 2]
```diff
@@ -1,16 +1,14 @@
 #!/bin/bash
 
-ENVTEST_K8S_VERSION=${ENVTEST_K8S_VERSION-1.28.3}
```
Comment (raulcabello) [inline tests/controllers/run_controller_tests.sh:3]: How can we provide the k8s version now? Would it make sense to add it as an env var in the github workflow? Like we do for [fleet](https://github.com/rancher/fleet/blob/main/.github/workflows/ci.yml#L39)

File: .github/workflows/controller-test.yml (hunk 1 above)
Comment (JonCrowther) [reply .github/workflows/controller-test.yml]: Removed

File: tests/controllers/run_controller_tests.sh (hunk 2 above)
Comment (JonCrowther) [reply tests/controllers/run_controller_tests.sh:3]: I re-added the environment variable in the script, and specified it in the workflow. I also added a line to the README on how to use it.

**Review by andreas-kupries — APPROVED** [review]

**Review by raulcabello — APPROVED** [review]

**Pull request merged**